strawberryfields/tdm/tdmprogram.py

```diff
@@ -335,12 +343,12 @@ def construct_circuit(self):
         # q[sm[2]] as concurrent modes of spatial mode C
         # q[sm[3]] as concurrent modes of spatial mode D.
 
-        for cmd in cmds:
+        for cmd in self.rolled_circuit:
             if isinstance(cmd.op, ops.Measurement):
                 self.measured_modes.append(cmd.reg[0].ind)
 
         for i in range(self.total_timebins):
-            for cmd in cmds:
+            for cmd in self.rolled_circuit:
                 self.apply_op(cmd, q, i)
```

Comment:
Would `self.rolled_circuit` ever be changed here? I mean via the `cmd`s somehow? I don't think so, but it might be good to double check since we're not using copies here.